papers/scipy/scipy.rst

```diff
@@ -0,0 +1,1706 @@
+                        p_k = 2*b_k, 1 <= k <= R
+
+A finite impulse response filter is basically a weighted moving
+average.  Given an input sequence :math:`{x_{_n}}` and the :math:`M+1`
+filter coefficient :math:`\{b_{_0}, \ldots, b_{_M}\}`, the filtered
+output :math:`{y_{_n}}` is computed as discrete convolution of
+:math:`x` and :math:`b`:
+
+.. math::
+   :label: eq-fir-filter
+
+   y_{_n} = \sum_{i=0}^{M} b_{_i}x_{_{n-i}}
+
+:math:`M` is the *order* of the filter; a filter with order :math:`M`
+has :math:`M + 1` coefficients.  It is common to say that the filter has
+:math:`M + 1` *taps*.
+
+Apply a FIR filter: convolution
+-------------------------------
+
+To apply a FIR filter to a signal, we use one of the convolution functions
+available in NumPy or SciPy, such as ``scipy.signal.convolve``.  For example,
+
+.. code-block:: python
+
+    from scipy.signal import convolve
+
+    # Make a signal to be filtered.
+    np.random.seed(123)
+    x = np.random.randn(50)
+    # taps is the array of FIR filter coefficients.
+    taps = np.array([ 0.0625,  0.25  ,  0.375 ,
+                      0.25  ,  0.0625])
+    # Filtered signal.
+    y = convolve(x, taps)
+
+There are also convolution functions in ``scipy.ndimage``.
+The function ``scipy.ndimage.convolve1d`` provides an ``axis`` argument,
+which allows all the signals stored in one axis of a multidimensional array
+to be filtered with one call.  For example,
+
+.. code-block:: python
+
+    from scipy.ndimage import convolve1d
+
+    # Make an 3-d array containing 1-d signals
+    # to be filtered.
+    x = np.random.randn(3, 5, 50)
+    # Apply the filter along the last dimension.
+    y = convolve1d(x, taps, axis=-1)
+
+**Edge effects/boundary conditions.**
+
+[TODO: Expand these comments on edge effects.]
+
+If one applies a FIR filter with
+one of the convolution functions, how to handle the edges
+must be decided.  Options include using the ``mode`` argument
+of the convolution function, or simply discarding values near
+the edge.
+
+Specialized functions that are FIR filters
+------------------------------------------
+
+..
+    TODO: either expand or delete this section.
+
+The uniform filter and the Gaussian filter implemented in ``scipy.ndimage``
+are FIR filters.  In the case of one-dimensional time series, the specific
+functions are ``uniform_filter1d`` and ``gaussian_filter1d``.
+
+The Savitzky-Golay filter [SavGol]_ is also a FIR filter. In the module
+``scipy.signal``, SciPy provides the function ``savgol_coeffs`` to create
+the coefficients of a Savitzy-Golay filter.  The function ``savgol_filter``
+applies the Savitzky-Golay filter to an input signal without returning the
+filter coefficients.
+
+FIR filter frequency response
+-----------------------------
+
+The function ``scipy.signal.freqz`` computes the frequency
+response of a FIR filter (and, more generally, an IIR filter represented
+by the coefficients of the numerator and denominator of the transfer
+function).
```

Comment:
Not sure I get the point of the part in brackets. Raises user questions like "why is IIR more general" and "why does it matter that that filter is in num/den format?

Reply:
Edited to be clearer (I hope!).